### DISCUSSION TITLE
Catch error for orm

### DISCUSSION THREAD

**Discussion by birdwyx** [vlang/v#15865] [General]
How can I catch an error happens in the statement below? 

the insert statement may fail due to duplicate key etc but there seems no way to catch this error?
sql db {
	insert new_customer into Customer
}

**Comment by enghitalo** [comment]
> How can I catch an error happens in the statement below? 
> 
> the insert statement may fail due to duplicate key etc but there seems no way to catch this error?
> sql db {
> 	insert new_customer into Customer
> }
```v
sql db {
    insert new_customer into Customer
} or {
   eprintln(err)
}
```

**Reply by birdwyx** [reply]
I tried the following code
```
        sql db {
                insert cartoon into Cartoon
        } or {
                panic("error")
        }
```

with V version 0.3.0 e5c53cf , I got the following error at compile time:

PS D:\ProjectsMy\vtoon\src> v run tdb.v
```
tdb.v:86:11: error: invalid expression: unexpected keyword `or`
   84 |         sql db {
   85 |                 insert cartoon into Cartoon
   86 |         } or {
      |           ~~
   87 |                 panic("error")
   88 |         }
```

while the following code can be successfully compiled :
```
        sql db {
                insert cartoon into Cartoon
        } 
```

I am thinking if I am at the right V version or if this is really supported on Win

PS D:\ProjectsMy\vtoon\src> v --version
V 0.3.0 e5c53cf

**Reply by enghitalo** [reply]
Hi, @birdwyx.
Well. This is a new feature.
https://github.com/vlang/v/pull/14989

Try `v up`

**Reply by JalonSolov** [reply]
Definitely do `v up`... and if that doesn't work, do `git pull` then `.\make.bat`.

Current V version is:
```
V 0.3.1 be67411
```

**Reply by birdwyx** [reply]
Thanks enghitalo and JalonSolov,  v up and fixed the problem
